keps/sig-cloud-provider/0003-testgrid-conformance-e2e.md

```diff
@@ -0,0 +1,270 @@
+---
+kep-number: 003
+title: Reporting Conformance Test Results to Testgrid
+authors:
+  - "@andrewsykim"
+owning-sig: sig-cloud-provider
+participating-sigs:
+  - sig-testing
+  - sig-release
+  - sig-aws
+  - sig-azure
+  - sig-gcp
+  - sig-ibmcloud
+  - sig-openstack
+  - sig-vmware
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2018-06-06
+last-updated: 2018-06-06
+status: provisional
+
+---
+
+# Reporting Conformance Test Results to Testgrid
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+
+## Summary
+
+This is a KEP outlining the motivation behind why cloud providers should periodically upload E2E conformance test results to [Testgrid](https://github.com/kubernetes/test-infra/tree/master/testgrid) and how a cloud provider can go about doing this.
+
+## Motivation
+
+The primary motivation behind collecting conformance test results from various cloud providers on a regular basis is to inform sig-release of any critical bugs. It's important to collect results from various cloud providers to increase coverage and inform other cloud providers of bugs they may be impacted by.
+
+### Goals
+
+* All SIGs / Subprojects owners representing a cloud provider is aware of the importance of frequently uploading conformance test results to testgrid
+* There is a clear and detailed process in place for any cloud provider to upload conformance test results to Testgrid.
+
+### Non-Goals
+
+* Test coverage - increasing test coverage is outside the scope of this KEP
+* CI/CD - what CI/CD tool is used to run E2E tests and upload results is outside the scope of this KEP. It is up to the cloud provider to decide where/how tests are run.
+* Cluster Provisioning - how a cluster is provisioned is outside the scope of this KEP.
+
+## Proposal
+
+We would like to propose that every Kubernetes cloud provider reports conformance test results for every patch version of Kubernetes at the minimum. Running conformance tests against master and on pre-release versions are highly encouraged but will not be a requirement.
+
+### Implementation Details/Notes/Constraints
+
+Before continuing, it is highly recommended you read the following documentation provided by sig-testing:
+* [Testgrid Configuration](https://github.com/kubernetes/test-infra/tree/master/testgrid#testgrid)
+* [Display Conformance Tests with Testgrid](https://github.com/kubernetes/test-infra/blob/master/testgrid/conformance/README.md)
+
+#### How to Run E2E Conformance Tests
```

Comment:
I'm assuming the goal of this is to find issues with cloud provider implementation specific stuff.  Do you need a bit more information about the set of binaries that the cloud provider should ensure are under test here (at least CCM, but are CSI impls included?  Anything else?)

Reply:
👍